### ISSUE TITLE
Allow plain format for standard error

### ISSUE THREAD

**Issue by msftrubengu** [PowerShell/DSC#286]
### Summary of the new feature / enhancement

The stardard error will contain all the format codes which is great seeing in a console, but will be hard to read in other scenarios. For example, 

```
"  \x1b[2m2023-11-29T00:30:07.457732Z\x1b[0m \x1b[32m INFO\x1b[0m \x1b[1;32mdsc\x1b[0m\x1b[32m: \x1b[32mReading input from STDIN\x1b[0m\n    \x1b[2;3mat\x1b[0m src\\main.rs:65\n\n  \x1b[2m2023-11-29T00:30:11.807947Z\x1b[0m \x1b[32m INFO\x1b[0m \x1b[1;32...
```

If those messages ends up in a log file, it will be really hard to follow up what is going on without the caller making some preprocessing.

This also happen in some consoles. For example, try using it from Windows Sandbox without installing Terminal.

### Proposed technical implementation details (optional)

From what I seen, there's already something implemented that sets the output to json without the need of `-f json` if not using `-i`. If it's a feature (maybe that's a bug?) then stdErr should follow something similar or there should be a new global option that allows plain formatting.

**Comment by anmenaga** [comment]
This seems to be a duplicate of #273 which is schedules to be fixed in next release.
To be sure, @msftrubengu please provide info where and how do you run dsc when you see these codes? (what OS; in the terminal or from other app)

**Comment by msftrubengu** [comment]
Yes, that is the same issue. I see it in Win11 calling dsc programmatically via [CreateProcess](https://github.com/microsoft/winget-cli/blob/b6bc2d9836456abb603dc4b279667be369b1c538/src/Microsoft.Management.Configuration.Processor.Dsc/DscProcessor.cpp#L87C13-L87C13)

I also see it using Windows PowerShell in Windows Sandbox running on Win11. In that scenario the stdOut also has formatting issues.

![image](https://github.com/PowerShell/DSC/assets/36937303/c5f34bb8-d372-49f1-9111-832726b955eb)

**Comment by SteveL-MSFT** [comment]
We probably want a commandline switch `--trace-format` that can simply return traces as JSON or plaintext (no VT).
